shared/reducers/chat2.js

```diff
@@ -265,24 +265,30 @@ const messageMapReducer = (messageMap, action, pendingOutboxToOrdinal) => {
         return message.set('transferState', null)
       })
     case Chat2Gen.attachmentDownload:
-      return messageMap.updateIn([action.payload.conversationIDKey, action.payload.ordinal], message => {
-        if (!message || message.type !== 'attachment') {
-          return message
+      return messageMap.updateIn(
+        [action.payload.message.conversationIDKey, action.payload.message.ordinal],
+        message => {
+          if (!message || message.type !== 'attachment') {
+            return message
+          }
+          return message.set('transferState', 'downloading')
         }
-        return message.set('transferState', 'downloading')
-      })
+      )
     case Chat2Gen.attachmentDownloaded:
-      return messageMap.updateIn([action.payload.conversationIDKey, action.payload.ordinal], message => {
-        if (!message || message.type !== 'attachment') {
+      return messageMap.updateIn(
+        [action.payload.message.conversationIDKey, action.payload.message.ordinal],
+        message => {
+          if (!message || message.type !== 'attachment') {
+            return message
+          }
+          const path = action.error ? '' : action.payload.path
           return message
+            .set('downloadPath', path)
```

Comment:
prefer using merge in this type of case

Reply:
Ok, all I did was change the indent here, I didn't actually change this.